syntax = "proto3";

package network;

message PBNetworkStop {
  double latitude = 1;
  double longitude = 2;
}

message PBNetworkStation {
  repeated PBNetworkStop stops = 1;
}

message PBNetworkStopTime {
  uint32 stop_id = 1;
  uint32 arrival_time = 2;
  uint32 departure_time = 3;
}

message PBNetworkTrip {
  repeated PBNetworkStopTime stop_times = 1;
}

message PBNetworkPath {
  uint32 from_stop_id = 1;
  uint32 to_stop_id = 2;
  uint32 duration = 3;
}

message PBNetwork {
  repeated PBNetworkStation stations = 1;
  repeated PBNetworkTrip trips = 2;
  repeated PBNetworkPath paths = 3;
}
